pkg/build/api/types.go

```diff
@@ -476,3 +479,12 @@ type BuildLogOptions struct {
 	// is not available yet. Otherwise the server will wait until the build has started.
 	NoWait bool
 }
+
+// SecretSpec specifies a secret to be included in a build pod and its corresponding mount point
+type SecretSpec struct {
```

Comment:
I would vote for not duplicating the pod template. If you want a full-blown pod, just craft one and set up your system to allow you to run it if you need privileged mode, etc. 

Allowing you to attach arbitrary volumes to custom builders could make it possible to mount external volumes and other cool stuff, but so far I don't know of a use case for it. The use case that we have right now for additional secrets is the custom builds for RCM that need additional secrets. My vote would be to scope additional secrets to only custom builds. Thinking about the mount point... I would even be more prescriptive and mount the secrets at a predetermined path based on the secret name, similar to how we mount the push/pull secrets. If you're already using a custom builder, you can set up your code to find the secret and copy it / use it any way it likes.